File: examples/js/timeframe.js
import * as qp from "qpace";

// Constructor
{
  const tf = qp.Timeframe.days(1);
  const tf = qp.Timeframe.hours(4);
  const tf = qp.Timeframe.minutes(15);
  const tf = qp.Timeframe.unknown(30);
}

// From string
{
  const tf = qp.Timeframe.fromString("1D");
  const tf = qp.Timeframe.fromString("4H");
  const tf = qp.Timeframe.fromString("15m");
  const tf = qp.Timeframe.fromString("?");
}

// To string
{
  const text = qp.Timeframe.days(1).toString();
  const text = qp.Timeframe.hours(4).toString();
  const text = qp.Timeframe.minutes(15).toString();
  const text = qp.Timeframe.unknown().toString();
}

File: examples/python/timeframe.py
import qpace as qp

# Constructor
if True:
    tf = qp.Timeframe.Days(1)
    tf = qp.Timeframe.Hours(4)
    tf = qp.Timeframe.Minutes(15)
    tf = qp.Timeframe.Unknown()

# From string
if True:
    tf = qp.Timeframe.from_str("1D")
    tf = qp.Timeframe.from_str("4H")
    tf = qp.Timeframe.from_str("15m")
    tf = qp.Timeframe.from_str("?")

# To string
if True:
    text = str(qp.Timeframe.Days(1))
    text = str(qp.Timeframe.Hours(4))
    text = str(qp.Timeframe.Minutes(15))
    text = str(qp.Timeframe.Unknown())
